src/Sylius/Component/Core/Promotion/Action/FixedDiscountPromotionActionCommand.php

```diff
@@ -59,12 +59,18 @@ public function execute(PromotionSubjectInterface $subject, array $configuration
             return;
         }
 
-        $this->isConfigurationValid($configuration);
+        $channelCode = $subject->getChannel()->getCode();
+        if (!isset($configuration[$channelCode])) {
+            return;
```

Comment:
I think we should not - if there is no configuration for channel it simple means there should be no discount on that channel. It's not necessarily an error.